Inital working flake implementation

nrdxp opened the pull request:
Resolves #174 by allowing those migrating to flakes to continue using NUR without breaking backwards compatibility for those that aren't.

Some packages or modules may be broken if they try to import anything from the
`NIX_PATH`. Many packages, overlays and modules work just fine without modification so this should still be quite useful.

Mic92 commented on flake.lock:
```diff
@@ -0,0 +1,11 @@
+            "inputs": {},
+            "narHash": "sha256-4p9gqMJTVmyH7eSq1euGZWusldWoBfKfi/VwZEPL2Mw=",
+            "originalUrl": "nixpkgs",
+            "url": "github:nrdxp/nixpkgs/1e232452ed4c57e9968f4342d9ba044d36403928"
```
I assume there is no way to import the users nixpkgs instead of this?

Mic92 replied:
Also this should point to NixOS instead of your fork.

Mic92 replied:
Ah you fixed some stuff: https://github.com/nrdxp/nixpkgs/commits/1e232452ed4c57e9968f4342d9ba044d36403928

nrdxp replied:
Well I didn't mean to use my fork. I have my fork pinned globally so it got set since I didn't specify the url. You can easily replace it in by specifying a flake uri explicitly, via `inputs.nixpkgs.url = "github:owner/nixpkgs/branch`, and then running `nix flake update`. I fix this and set it to the default upstream since it was a mistake to begin with.

Mic92 commented:
Since I try to document every possible feature of NUR: Can you add some documentation on how to use the flake in the README?

nrdxp commented:
Yeah sure, I should have some time this evening to get that done

nrdxp commented:
Sorry about the delay on this, I had some personal stuff come up and I forgot I had this open. Rebased on latest changes and updated README.md as per request. Also fixed `flake.lock` to point to upstream instead of my fork.

Mic92 commented on flake.lock:
```diff
@@ -0,0 +1,11 @@
+            "inputs": {},
+            "narHash": "sha256-OnpEWzNxF/AU4KlqBXM2s5PWvfI5/BS6xQrPvkF5tO8=",
+            "originalUrl": "nixpkgs",
+            "url": "github:edolstra/nixpkgs/7f8d4b088e2df7fdb6b513bc2d6941f1d422a013"
```
Should this not be the NixOS one?

```suggestion
            "url": "github:NixOS/nixpkgs/7f8d4b088e2df7fdb6b513bc2d6941f1d422a013"
```

zimbatm replied:
nixpkgs hasn't gained flake support yet

bqv replied:
It has now

Mic92 commented on flake.nix:
```diff
@@ -0,0 +1,15 @@
+      overlay = final: prev: {
+        nur = import ./default.nix {
+          nurpkgs = nixpkgs;
+          pkgs = nixpkgs.pkgs;
```
This effectively pin-points the nur to the nixpkgs version used in the repository?

bqv commented:
Bump. Now that I've switched my setup to flakes, I'd appreciate it if this was merged even if only into a branch, so I don't have to maintain my separate fork (and predumably @nrdxp can delete his too)

Mic92 commented:
My question is still not answered. To me it seems as if this would pin point nixpkgs. If that is the case we need a mechanism to update our nixpkgs revision. 
A separate branch is not useful at all because we only update other branches but master.

bqv commented:
I don't quite understand what you're asking, but if it helps, that nixpkgs input is used only for the stuff entirely internal to NUR, so you really shouldn't have to update it regularly (or even at all), unless you're rapidly changing stuff within this repo itself.
E.g. checking my flake.lock currently, i see there's a nixpkgs that I pull in for my system, there's the old revision for the nur, and there's another revision pulled in by home-manager.

There is no reason to have a fancy mechanism here, you can just add the flake.nix and flake.lock

Mic92 commented:
Ok. How would I add packages from nix flakes than?

bqv commented:
In what context? Nothing needs to change in the NUR repository

Mic92 commented:
After having a conversation with @bqv I realised we indeed just can use it as it is.
